homeassistant/components/proximity/coordinator.py

```diff
@@ -164,16 +164,20 @@ def _calc_distance_to_zone(
             )
             return None
 
-        distance_to_zone = distance(
+        distance_to_centre = distance(
             zone.attributes[ATTR_LATITUDE],
             zone.attributes[ATTR_LONGITUDE],
             latitude,
             longitude,
         )
 
         # it is ensured, that distance can't be None, since zones must have lat/lon coordinates
-        assert distance_to_zone is not None
-        return round(distance_to_zone)
+        assert distance_to_centre is not None
+
+        zone_radius: float = zone.attributes["radius"]
+        distance_to_edge: int = round(distance_to_centre - zone_radius)
+        # return 0 when we've arrived the zone
+        return max(distance_to_edge, 0)
```

Comment:
```suggestion
        if zone_radius > distance_to_centre:
            # return 0 when we've arrived the zone
            return 0
        return round(distance_to_centre - zone_radius)
```

this avoids the calculation and rounding of the distance, when we've already arrived at the zone

Reply:
Looks good. 👍I am undecided if it actually is a worthwhile optimization but it's probably even less worthwhile to overthink it. :D